src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests4.cs

```diff
@@ -29,12 +29,9 @@ public void M<T>(T t)
     }
 }");
             comp.VerifyDiagnostics(
-                // (8,18): error CS0150: A constant value is expected
-                //             case default:
-                Diagnostic(ErrorCode.ERR_ConstantExpected, "default").WithLocation(8, 18),
-                // (8,18): error CS8313: A default literal 'default' is not valid as a case constant. Use another literal (e.g. '0' or 'null') as appropriate. If you intended to write the default label, use 'default:' without 'case'.
+                // (8,18): error CS8716: No target type was found for the default literal.
```

Comment:
>// (8,18): error CS8716: No target type was found for the default literal. [](start = 16, length = 74)

Isn't there a target type - T?

Reply:
What happens when T has a class constraint?

---
In reply to: [314127487](https://github.com/dotnet/roslyn/pull/37596#discussion_r314127487) [](ancestors = 314127487)